File: spring/jpa/src/main/java/com/example/jpa/cashing/Account.java
package com.example.jpa.cashing;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Objects;
import java.util.UUID;

@Entity
public class Account {

    @Id
    public UUID id;
    public UUID organizationId;

    public String name;

    public Account() {
        id = UUID.randomUUID();
    }

    public Account(UUID id, UUID organizationId, String name) {
        this.id = id;
        this.organizationId = organizationId;
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Account account = (Account) o;
        return Objects.equals(id, account.id) && Objects.equals(organizationId, account.organizationId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, organizationId);
    }
}

File: jpa-postgre-spring-kotlin/src/main/kotlin/com/example/jpapostgrespringkotlin/Account.kt
package com.example.jpapostgrespringkotlin

import com.vladmihalcea.hibernate.type.json.JsonBinaryType
import java.util.UUID
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table
import org.hibernate.annotations.Type
import org.hibernate.annotations.TypeDef


@Entity
@Table(name = "account")
@TypeDef(
    name = "jsonb",
    typeClass = JsonBinaryType::class
)
class Account(

    @Column(name = "name")
    var name: String? = null,

    @Type(type = "jsonb")
    @Column(name = "history", columnDefinition = "jsonb")
    var history: Any? = null,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null
}
